packages/termux-tools/pkg

```diff
@@ -18,8 +18,50 @@ show_help() {
 	exit 1
 }
 
-check_pkgcache() {
-	if [ -z "$(find @TERMUX_PREFIX@/var/cache/apt/pkgcache.bin -mmin -5)" ]; then
+select_mirror() {
+	local main_repo="https://dl.bintray.com/termux/termux-packages-24"
+	local top_mirrors="${main_repo} https://termux.mentality.rip/termux-packages-24 https://grimler.se/termux-packages-24 https://main.termux-mirror.ml"
+
+	local current_mirror
+	current_mirror=$(grep -P "^\s*deb\s+" @TERMUX_PREFIX@/etc/apt/sources.list | grep -oP 'https?://[a-z0-9/._-]+')
+
+	# Do not update mirror if:
+	# * Uses .cn domain - specific to Chinese users.
+	# * If $TERMUX_PKG_NO_MIRROR_SELECT was set.
+	if [ -n "${TERMUX_PKG_NO_MIRROR_SELECT-}" ] || grep -qP '.+\.cn/' <(echo "${current_mirror-x}"); then
+		return
+	fi
+
+	if [ -n "$(find @TERMUX_PREFIX@/var/cache/apt/pkgcache.bin -mmin -30 2>/dev/null)" ]; then
+		# Mirrors are rotated if either they are not working or 30 minutes timeout
+		# has passed.
+		if [ -n "${current_mirror}" ] && curl --head --fail --location "${current_mirror}/dists/stable/Release" >/dev/null 2>&1; then
+			echo "Reusing mirror: ${current_mirror}"
+			return
+		fi
+	fi
+
+	# Shuffle mirrors and pick the first one which is accessible.
+	local selected_mirror m
+	for m in $(echo "$top_mirrors" | tr ' ' '\n' | shuf); do
```

Comment:
Btw, `shuf` is in the `debianutils` package, which isn't installed for me. I didn't uninstall it either, I just have a Termux that hasn't been reinstalled since this repo dropped Android 5/6, only updated, and the recent userland changes didn't force installing `debianutils`.

Reply:
> Btw, shuf is in the debianutils package

dpkg thinks differently - `shuf` is part of coreutils package.

Reply:
coreutils docs also show that: https://www.gnu.org/software/coreutils/manual/html_node/shuf-invocation.html

Reply:
My mistake, I misread this command output: :laughing:
```
> which shuf
The program which is not installed. Install it by executing:
 pkg install debianutils
```